plugins/state_history_plugin/state_history_plugin.cpp

```diff
@@ -296,6 +298,16 @@ struct state_history_plugin_impl : std::enable_shared_from_this<state_history_pl
    std::map<session*, std::shared_ptr<session>> sessions;
 
    void listen() {
+      if ( app().is_plugin_initialized<history_plugin>() ) {
+         wlog( "Legacy history plugin is running; this will cause random crashes." );
+         if ( allow_legacy_history_plugin ) {
+            wlog( "With state-history-allow-legacy-history-plugin set, proceed at your own risk" );
+         } else {
+            wlog( "With state-history-allow-legacy-history-plugin not set, shutting down." );
```

Comment:
I'm afraid this message will encourage people to set the flag. Also, since it's shutting down, use `elog`.

Maybe "Stopping to prevent random crashes; disable the legacy history plugin to continue". People who are desperate to do the wrong thing will find the flag without you mentioning it.